### PR TITLE
Use `athena` for submitting RUM events

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/18033
Needs https://github.com/giantswarm/config/pull/479 and https://github.com/giantswarm/athena/pull/43 for testing.

This PR changes the RUM event reporting logic, so it would use `athena` instead of `api`. It contains a GraphQL client, logic for interacting with `athena` and a bit of refactoring of the way we triggered RUM events.

Functionality should be the same as before.

### CODE REVIEW

File: src/@types/graphql.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+declare module '*.graphql' {
```
Comment (axbarsan) [inline src/@types/graphql.d.ts:1]: So TS knows that we get strings from `.graphql` files

File: src/@types/realUserMonitoring.d.ts [hunk 2]
```diff
@@ -1,11 +0,0 @@
-interface IRum {
```
Comment (axbarsan) [inline src/@types/realUserMonitoring.d.ts:1]: These are leftovers from when we used DataDog

File: src/model/clients/GraphQLClient.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export interface IGraphQLClient {
+  executeQuery<T = unknown>(
+    query: string,
+    variables?: GraphQLVariables
+  ): Promise<T>;
+}
+
+export class GraphQLClientImpl implements IGraphQLClient {
```
Comment (axbarsan) [inline src/model/clients/GraphQLClient.ts:41]: Didn't want to add another big-ass dependency for this

File: webpack.common.ts [hunk 4]
```diff
@@ -96,6 +96,11 @@ const config: webpack.Configuration = {
           mimetype: 'image/svg+xml',
         },
       },
+      {
+        test: /\.(graphql|gql)$/,
```
Comment (axbarsan) [inline webpack.common.ts:100]: This is so when we import from `.graphql` files, they actually get embedded in the app as strings

File: src/model/services/athena/mutations/CreateAnalyticsEvent.graphql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+mutation CreateAnalyticsEvent($input: AnalyticsEventInput!) {
```
Comment (axbarsan) [inline src/model/services/athena/mutations/CreateAnalyticsEvent.graphql:1]: The GraphQL mutation for submitting an event

**Review by marians — APPROVED** [review]

**Review by kuosandys — APPROVED** [review]

**Pull request merged**